lib/types/query/equality.ts

```diff
@@ -1,9 +1,9 @@
 import { BasicEntryField, EntryFields } from '../entry'
 import { ConditionalQueries } from './util'
 
-// TODO: Note: Equality and inequality operators are not supported for text fields
-// What types do we hav to exclude here?
-type SupportedTypes = Exclude<BasicEntryField, EntryFields.RichText> | undefined
+type SupportedTypes =
+  | Exclude<BasicEntryField, EntryFields.Object | EntryFields.RichText | EntryFields.Location>
```

Comment:
Are we sure this works? If one of the excluded types is a union (like `Object`) it means excluding all of the types in the union.